app/services/trainees/create_from_hesa.rb

```diff
@@ -107,14 +107,18 @@ def course_attributes
         course_min_age: age_range && age_range[0],
         course_max_age: course_max_age,
         study_mode: study_mode,
-        itt_start_date: hesa_trainee[:itt_start_date],
+        itt_start_date: trainee_start_date,
         itt_end_date: hesa_trainee[:itt_end_date],
-        trainee_start_date: hesa_trainee[:trainee_start_date] || hesa_trainee[:itt_start_date],
+        trainee_start_date: trainee_start_date,
       }
 
       primary_education_phase? ? fix_invalid_primary_course_subjects(attributes) : attributes
     end
 
+    def trainee_start_date
+      hesa_trainee[:itt_commencement_date].presence || hesa_trainee[:commencement_date]
```

Comment:
looks good - I wonder if the code might benefit from a comment here as the meaning of itt_commencement_date and commencement_date are not obvious. Something like:

```
# Use itt_commencement_date first, this is populated when the trainee has transferred from a non-QTS 
# awarding course, to an ITT (QTS awarding) course, otherwise use commencement_date. 
# This is the start date for trainees who have not transferred courses.
```

Reply:
Agreed would be good for a comment about why we're remapping these.

Reply:
Done!